libs/beacon/src/beacon_setup_service.cpp

```diff
@@ -145,32 +153,70 @@ BeaconSetupService::State BeaconSetupService::OnIdle()
     }
     else
     {
-      // Initiating setup
-      std::set<MuddleAddress> cabinet;
-      for (auto &m : beacon_->aeon.members)
-      {
-        cabinet.insert(m.identifier());
-      }
-
-      pre_dkg_rbc_.ResetCabinet(cabinet);
-      rbc_.ResetCabinet(cabinet);
-      auto cabinet_size = static_cast<uint32_t>(cabinet.size());
-      complaints_manager_.ResetCabinet(cabinet_size);
-      complaints_answer_manager_.ResetCabinet(cabinet_size);
-      qual_complaints_manager_.Reset();
-
-      return State::WAIT_FOR_DIRECT_CONNECTIONS;
+      SetTimeToProceed(State::RESET);
+      return State::RESET;
     }
   }
 
   state_machine_->Delay(std::chrono::milliseconds(100));
   return State::IDLE;
 }
 
-BeaconSetupService::State BeaconSetupService::OnWaitForDirectConnections()
+/**
+ * Reset and initial state for the DKG. It should return to this state in the
+ * case of total DKG failure
+ *
+ */
+BeaconSetupService::State BeaconSetupService::OnReset()
 {
   std::lock_guard<std::mutex> lock(mutex_);
-  dkg_state_gauge_->set(static_cast<uint8_t>(State::WAIT_FOR_DIRECT_CONNECTIONS));
+  beacon_dkg_state_gauge_->set(static_cast<uint8_t>(State::RESET));
+
+  assert(beacon_);
+
+  // Initiating setup
+  std::set<MuddleAddress> cabinet;
+  for (auto &m : beacon_->aeon.members)
+  {
+    cabinet.insert(m.identifier());
+  }
+
+  auto cabinet_size = static_cast<uint32_t>(cabinet.size());
+
+  coefficients_received_.clear();
+  complaints_answer_manager_.ResetCabinet(0);
+  complaints_answer_manager_.ResetCabinet(cabinet_size);
+  complaints_manager_.Clear();
+  complaints_manager_.ResetCabinet(0);
+  complaints_manager_.ResetCabinet(cabinet_size);
```

Comment:
complaints_manager_.Clear() and ResetCabinet(0) is unnecessary here if you call ResetCabinet(cabinet_size)